.github/workflows/build_env_run_tests.yml

```diff
@@ -0,0 +1,56 @@
+# Builds the aeon environment, flakes via flake8, checks type annotations via mypy, tests via pytest, 
+# reports test coverage via pytest-cov and codecov, and reports security vulnerabilities via bandit.
+
+name: build_env_run_tests
+on:
+  push:
+    branches: [ main, reorg, jai_dev, config ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:  # Allows running manually from Github's 'Actions' tab
+jobs:
+  build_env_run_tests:
+    name: Build env and run tests on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
```

Comment:
different python versions?

Reply:
Yea possibly, I guess it depends how extensive we want this to be.

Reply:
It was actually a bit expensive to run for multiple python versions, so I've specified only 3.11 here

Reply:
Do you know if python versioning numbers roughly follow semver (I somehow suspect they don't)? If they did, and python 3.11 was a strict superset of 3.9 it would be best here to actually run tests against python 3.9 only.

Reply:
They don't strictly follow semver, so there is a chance there are regressions... 

We basically decided today to just support one version, in this case 3.11 - are you fine with that @glopesdev ?